### PR TITLE
Javadoc for ResponderServer

### PR DESCRIPTION
**Pull request by anstr**

### CODE REVIEW

File: core/src/main/java/io/github/tcdl/Responder.java [hunk 1]
```diff
@@ -12,10 +12,10 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * {@link Responder} is a component which holds both original request message and response message.
- * Is used by {@link ResponderServer} to send responses and acknowledgements
+ * {@link Responder} is a component which holds original request message and create response base on it.
+ * Is used by {@link ResponderServer} to send responses and acknowledgements.
  *
- * Created by rdro on 4/29/2015.
+ * @author Roman Drozdov
```
Comment (vso-tc) [inline core/src/main/java/io/github/tcdl/Responder.java:18]: Isn't it better to specify github user id instead of real name?

Comment (rkatsyuryna) [inline core/src/main/java/io/github/tcdl/Responder.java:18]: or delete this line as agreed

File: core/src/main/java/io/github/tcdl/MsbContext.java [hunk 2]
```diff
@@ -72,11 +70,19 @@ public TimeoutManager getTimeoutManager() {
         private boolean withShutdownHook;
         private boolean withDefaultChannelMonitorAgent;
 
+        /**
+         * Specifies if to shutdown current context during JVM exit. Is set to true will shutdown context regardless of
```
Comment (vso-tc) [inline core/src/main/java/io/github/tcdl/MsbContext.java:74]: Is -> If?

File: core/src/main/java/io/github/tcdl/MsbContext.java [hunk 3]
```diff
@@ -72,11 +70,19 @@ public TimeoutManager getTimeoutManager() {
...
+         * user will make a call to MsbContext.shutdown() from within client code.
+         */
         public MsbContextBuilder withShutdownHook(boolean withShutdownHook) {
             this.withShutdownHook = withShutdownHook;
             return this;
         }
 
+        /**
+         * Specifies if monitoring is enabled.
```
Comment (vso-tc) [inline core/src/main/java/io/github/tcdl/MsbContext.java:83]: monitoring -> monitoring agent

File: core/src/main/java/io/github/tcdl/Responder.java [hunk 4]
```diff
@@ -40,10 +39,10 @@ public Responder(MessageTemplate config, Message originalMessage, MsbContext msb
     /**
      * Send acknowledge message.
      *
-     * @param timeoutMs Time to wait for responsesRemaining
-     * @param responsesRemaining Expected number of responses
+     * @param timeoutMs time to wait for remaining responses
```
Comment (vso-tc) [inline core/src/main/java/io/github/tcdl/Responder.java:42]: let's specify that the timeout is in milliseconds

File: core/src/main/java/io/github/tcdl/Responder.java [hunk 5]
```diff
@@ -77,12 +81,11 @@ private void validateReceivedMessage(Message originalMessage) {
         Validate.notNull(originalMessage.getTopics(), "the 'originalMessage.topics' must not be null");
     }
 
+    /**
+     *
+     * @return original message which was received
```
Comment (vso-tc) [inline core/src/main/java/io/github/tcdl/Responder.java:86]: received -> sent?

File: core/src/main/java/io/github/tcdl/config/RequestOptions.java [hunk 6]
```diff
@@ -1,9 +1,10 @@
 package io.github.tcdl.config;
 
 /**
- * Options to configure a requester to wait for acknowledges or responses
- *
- * Created by rdrozdov-tc on 6/22/15.
+ * Options used while constructing {@link io.github.tcdl.Requester} that specify number and time to wait for acknowledgements or responses.
+ * Not all combinations make sense for this object.
+ * e.g: - setting {@literal waitForResponses<=0} will not activate timeout tasks
+ *      - {@literal responseTimeout > ackTimeout} will not activate ackTimeout task
```
Comment (vso-tc) [inline core/src/main/java/io/github/tcdl/config/RequestOptions.java:7]: Seems to be not correct

File: core/src/main/java/io/github/tcdl/MsbContext.java (hunk 2 above)
Comment (rkatsyuryna) [inline core/src/main/java/io/github/tcdl/MsbContext.java:74]: if

File: core/src/main/java/io/github/tcdl/MsbContext.java (hunk 3 above)
Comment (rkatsyuryna) [inline core/src/main/java/io/github/tcdl/MsbContext.java:83]: agree

**Pull request merged**